feature-libs/asm/components/asm-main-ui/asm-main-ui.component.ts

```diff
@@ -147,13 +148,44 @@ export class AsmMainUiComponent implements OnInit, OnDestroy {
               result.actionType === CustomerListColumnActionType.ACTIVE_CART
             ) {
               this.routingService.go({ cxRoute: 'cart' });
+            } else if (
+              'actionType' in result &&
+              result.actionType === CustomerListColumnActionType.CUSTOMER_360
+            ) {
+              this.customer$
+                .pipe(
+                  filter((curCustomer) => !!curCustomer),
+                  take(1)
+                )
+                .subscribe((customer) => {
+                  setTimeout(() => {
+                    this.showC360Dialog(customer);
+                  }, 500);
+                });
             }
           }
         })
     );
     this.subscribeForDeeplink();
   }
 
+  protected showC360Dialog(customer: User | undefined): void {
+    const data = { customer: customer };
+    this.launchDialogService?.openDialogAndSubscribe(
+      LAUNCH_CALLER.ASM_CUSTOMER_360,
+      this.element,
+      data
+    );
+
+    this.subscription.add(
+      this.launchDialogService?.dialogClose
```

Comment:
**?** is redundant.

Reply:
yes, we can also remove it.

I guess that we fristly add 'launchDialogService' to custructor with a breakchange, then to haddle the break change, we had to difine it as ```
constructor(
............
    protected launchDialogService**?**: LaunchDialogService
  ) {}
```
and in the major version, we don't remove all redundant code.

Reply:
Yes, I guess so